Model/Quote/Address/Total/Surcharge.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * @by SwiftOtter, Inc. 3/30/17
+ * @website https://swiftotter.com
+ **/
+
+namespace SwiftOtter\ShippingSurcharge\Model\Quote\Address\Total;
+
+use \SwiftOtter\ShippingSurcharge\Model\Surcharge as SurchargeModel;
+
+class Surcharge extends \Magento\Quote\Model\Quote\Address\Total\AbstractTotal
+{
+    /**
+     * @var \Magento\Framework\Pricing\PriceCurrencyInterface
+     */
+    private $priceCurrency;
+
+    /**
+     * @var \SwiftOtter\ShippingSurcharge\Api\Calculator\SurchargeCalculatorInterface
+     */
+    private $surchargeCalculator;
+
+    /**
+     * @var \Magento\Quote\Api\Data\TotalSegmentExtensionFactory
+     */
+    private $cartExtensionFactory;
+
+
+    public function __construct(
+        \Magento\Framework\Pricing\PriceCurrencyInterface $priceCurrency,
+        \SwiftOtter\ShippingSurcharge\Api\Calculator\SurchargeCalculatorInterface $surchargeCalculator,
+        \Magento\Quote\Api\Data\CartExtensionFactory $cartExtensionFactory
+    )
```

Comment:
The PSR on long constructors is to format them like this: `) {`. It takes a bit of getting used to, but I've come to like it.